src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -273,6 +275,105 @@ struct Filter
+        }
+
+        auto params = std::shared_ptr<AVCodecParameters>(avcodec_parameters_alloc(),
+                                                         [](AVCodecParameters* ptr) { avcodec_parameters_free(&ptr); });
+        if (!params) {
+            FF_RET(AVERROR(ENOMEM), "avcodec_parameters_alloc");
+        }
+        params->width      = mode->GetWidth();
+        params->height     = mode->GetHeight();
+        params->codec_type = AVMEDIA_TYPE_VIDEO;
+        params->codec_id   = AV_CODEC_ID_V210;
+        params->format     = AV_PIX_FMT_YUV422P10;
+
+        FF(avcodec_parameters_to_context(ctx.get(), params.get()));
+
+        // int thread_count = env::properties().get(L"configuration.ffmpeg.producer.threads", 0);
+        FF(av_opt_set_image_size(ctx.get(), "video_size", mode->GetWidth(), mode->GetHeight(), 0));
+        FF(avcodec_open2(ctx.get(), codec, nullptr));
+    }
+
+    std::shared_ptr<AVFrame> decode(IDeckLinkVideoInputFrame* video, const com_ptr<IDeckLinkDisplayMode>& mode)
+    {
+        void* video_bytes = nullptr;
+        if (SUCCEEDED(video->GetBytes(&video_bytes)) && video_bytes) {
+            video->AddRef();
+
+            auto frame = std::shared_ptr<AVFrame>(av_frame_alloc(), [video](AVFrame* ptr) {
+                video->Release();
+                av_frame_free(&ptr);
+            });
+            if (!frame)
+                FF_RET(AVERROR(ENOMEM), "av_frame_alloc");
+
+            if (hdr_) {
+                const auto size = video->GetRowBytes() * video->GetHeight();
+                AVPacket   packet;
+                av_init_packet(&packet);
+                packet.data = reinterpret_cast<uint8_t*>(video_bytes);
+                packet.size = size;
+                FF(avcodec_send_packet(ctx.get(), &packet));
+                FF(avcodec_receive_frame(ctx.get(), frame.get()));
+            } else {
+                frame->format      = AV_PIX_FMT_UYVY422;
+                frame->width       = video->GetWidth();
+                frame->height      = video->GetHeight();
+                frame->data[0]     = reinterpret_cast<uint8_t*>(video_bytes);
+                frame->linesize[0] = video->GetRowBytes();
+                frame->key_frame   = 1;
+            }
+
+            frame->interlaced_frame = mode->GetFieldDominance() != bmdProgressiveFrame;
+            frame->top_field_first  = mode->GetFieldDominance() == bmdUpperFieldFirst ? 1 : 0;
+
+            return frame;
+        }
+        return nullptr;
+    }
+};
+
+core::color_space get_color_space(IDeckLinkVideoInputFrame* video)
+{
+    IDeckLinkVideoFrameMetadataExtensions* md = nullptr;
+
+    if (SUCCEEDED(video->QueryInterface(IID_IDeckLinkVideoFrameMetadataExtensions, (void**)&md))) {
+        auto     metadata = wrap_raw<com_ptr>(md, true);
+        LONGLONG color_space;
+        if (SUCCEEDED(md->GetInt(bmdDeckLinkFrameMetadataColorspace, &color_space))) {
+            if (color_space == bmdColorspaceRec2020) {
```

Comment:
it will matter once that `color_space = is_hd ? params.pix_desc.color_space : core::color_space::bt601;` line in the mixer gets reworked with an auto-mode, or gets removed entirely.